src/dbnode/storage/series/series.go

```diff
@@ -391,6 +405,19 @@ func (s *dbSeries) LoadBlock(
 	block block.DatabaseBlock,
 	writeType WriteType,
 ) error {
+	switch writeType {
+	case WarmWrite:
+		at := block.StartTime()
+		alreadyExists, err := s.blockRetriever.IsBlockRetrievable(at)
```

Comment:
This seems to be hitting nil references in tests, it looks like `NewDatabaseSeries` does not set a `blockRetriever`

Reply:
Yup, I fixed this by updating all the tests with an opts struct too.